packages/framework-core/src/booster-event-dispatcher.ts

```diff
@@ -38,9 +38,16 @@ export class BoosterEventDispatcher {
 
   private static eventProcessor(eventStore: EventStore, readModelStore: ReadModelStore): EventsStreamingCallback {
     return async (entityName, entityID, eventEnvelopes, config) => {
-      // TODO: Separate the snapshot creation/read-model generation from the event handling into two independent processes.
-      await BoosterEventDispatcher.snapshotAndUpdateReadModels(config, entityName, entityID, eventStore, readModelStore)
-      await BoosterEventDispatcher.dispatchEntityEventsToEventHandlers(eventEnvelopes, config)
+      await Promise.allSettled([
+        await BoosterEventDispatcher.snapshotAndUpdateReadModels(
+          config,
+          entityName,
+          entityID,
+          eventStore,
+          readModelStore
+        ),
+        await BoosterEventDispatcher.dispatchEntityEventsToEventHandlers(eventEnvelopes, config),
+      ])
```

Comment:
With the changes introduced in this PR, and given that at this point all events are 100% persisted and settled in the database, it does no longer matter the order in which we calculate the read models and process event handlers, so we can now process them in parallel.